### ISSUE TITLE
Quicksilver Pushback Fails on BitBucket

### ISSUE THREAD

**Issue by ataylorme** [pantheon-systems/quicksilver-pushback#12]
First I ran into `Error parsing Bitbucket URL from Build Metadata` because the BitBucket repo URL in `build-metadata.json` was `http` and not `https`.

`push-back.php` [checks strictly for `https` with BitBucket](https://github.com/pantheon-systems/quicksilver-pushback/blob/master/push-back.php#L66) but is [more flexible with GitLab](https://github.com/pantheon-systems/quicksilver-pushback/blob/master/push-back.php#L53).

I'm not sure if we should make BitBucket more flexible here or modify Build Tools to ensure `https` only is written to `build-metadata.json`.

Once the was resolved, I ran into `fatal: could not read Username for 'https://bitbucket.org': No such device or address` in the pushback process when `git add .` is run.

@aaronbauman have you been using Quicksilver Pushback with BitBucket successfully? If so, I'm curious what is different in your configuration than mine.

**Comment by aaronbauman** [comment]
Do you have "user" defined in tokens.json?

My tokens.json looks like this, with `USERNAME` and `PASSWORD` replaced appropriately:
`{ "user": "USERNAME", "pass": "PASSWORD", "token": "dummy" }`

**Comment by ataylorme** [comment]
Ah, I authed with a token so mine looks like `{"token":"REDACTED_TOKEN_VALUE"}`

**Comment by aaronbauman** [comment]
yeah, bitbucket cloud rest api doesn't support access tokens, so you have to use username + password

**Comment by ataylorme** [comment]
Okay, that seems like a documentation issue then, alerting folks if they use a token Quicksilver pushback won't work. Maybe warn in the `README` and during `build:project:create` (if users have a saved token or enter a token instead of username/password)

Alternatively, we could remove token support and only support username/password for Build Tools.

@greg-1-anderson I'm curious what your thoughts are.

**Comment by aaronbauman** [comment]
Aren't we prompting for bb username/password already during build:project:create?
Would it be possible to inject these into tokens.json?

**Comment by ataylorme** [comment]
I have a saved token from months ago. I will clear that out and test `build:project:create` again to see what is prompted

**Comment by ataylorme** [comment]
Odd, when I look in `~/.terminus/cache/build-tools/` I see files for `BITBUCKET_USER` and `BITBUCKET_PASS`, neither of which have the entire token.

```
tree ~/.terminus/cache/build-tools/                                                                                             
├── <redacted_email>-ADMIN_EMAIL
├── <redacted_email>-ADMIN_PASSWORD
├── <redacted_email>-BITBUCKET_PASS
├── <redacted_email>-BITBUCKET_USER
├── <redacted_email>-CIRCLE_TOKEN
├── <redacted_email>-GITHUB_TOKEN
├── <redacted_email>-GITLAB_TOKEN
├── <redacted_email>-GIT_USER_EMAIL
└── <redacted_email>-SITE_NAME
```

Digging into the code for `build:project:create` it seems to set the `token`, derived from the username and password, in `tokens.json`. See https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/master/src/Commands/ProjectCreateCommand.php#L348:L350

Perhaps we should do as you suggest, and store `user` and `pass` entries as well

**Comment by aaronbauman** [comment]
Opened https://github.com/pantheon-systems/terminus-build-tools-plugin/pull/268 to address this.

**Comment by ataylorme** [comment]
@aaronbauman I can confirm this is working once I set `user` and `pass` in `tokens.json`. See https://bitbucket.org/ataylorme/example-wp-composer-pipelines-test-2019-09-10/pull-requests/2/wip-test-quicksilver-pushback/diff#Lweb/quicksilver-pushback-test.txtT2

**Comment by ataylorme** [comment]
https://github.com/pantheon-systems/terminus-build-tools-plugin/pull/268 makes it so `tokens.json` has entries for `token`, `user` and `pass` with BitBucket.

Quicksilver pushback still fails because of the `url` value in `build-metadata.json` starting with `http://bitbucket.org` and not `https`

**Comment by ataylorme** [comment]
Perhaps the BitBucket provider should use `alterBuildMetadata` to ensure that the `url` is `https`.

**Comment by ataylorme** [comment]
Actually, `build:project:create` set `url` to `git@bitbucket.org:ataylorme/example-wp-composer-pipelines-test-2019-09-11.git`

So I'm thinking it gets changed when a new push from BitBucket happens.

**Comment by ataylorme** [comment]
Yup - `[notice] Set build-metadata.json to {"url":"http://bitbucket.org/$BITBUCKET_USER/...`

**Comment by aaronbauman** [comment]
Looks like this is happening during initialization on bitbucket.
![messageagency___pipe-test___Pipelines_—_Bitbucket](https://user-images.githubusercontent.com/508451/64735687-0e910700-d4b7-11e9-83dc-de9f88e3329a.jpg)

**Comment by ataylorme** [comment]
@aaronbauman I opened https://github.com/pantheon-systems/terminus-build-tools-plugin/pull/270 to force `https` in `build-metadata.json` and am going to test that to see if it helps

**Comment by ataylorme** [comment]
Hmm, the changes in #270 didn't affect the `url` value in `build-metadata.json` sent to Pantheon. I'm going to bed now but will look at this again tomorrow
